### PR TITLE
Add coefficient of determination metric

### PR DESCRIPTION
**Pull request by slang03**
## What changes were proposed in this pull request?
Distinguish between Pearson Correlation and Coefficient of Determination as for the implementation of `correlationR2` actually calculated the Pearson Correlation. Additionally the `rSquared` method was added, which calculates the Coefficient of Determination.

Fixes: #4342

## How was this patch tested?
Test values were added in `RegressionEvalTest` and computed with [sklearn.metrics.r2_score](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html)

### CODE REVIEW

**Comment by slang03** [comment]
Additionally, the documentation [here](https://github.com/deeplearning4j/deeplearning4j/blame/gh-pages/evaluation.md#L128) has to be updated. The values for the "shellfish" dataset have to be computed. Does anyone have a link to that dataset?
